core/src/main/scala/cats/data/NonEmptyVector.scala

```diff
@@ -0,0 +1,179 @@
+package cats
+package data
+
+import scala.annotation.tailrec
+import scala.collection.immutable.VectorBuilder
+import cats.std.vector._
+
+/**
+ * A data type which represents a single element (head) and a Vector
+ * (tail).  This can be used to represent a Vector which is guaranteed
+ * to not be empty.
+ */
+final case class NonEmptyVector[A] private (vector: Vector[A]) {
+
+  def unwrap: Vector[A] = vector
+
+  def head: A = vector.head
+
+  def tail: Vector[A] = vector.tail
+
```

Comment:
alternatively, since `toVector` is free, we could have only `get` and require `ne.toVector(12)` for unsafe apply.